ui/src/routes/incarnations/DiffIncarnation.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useRef } from 'react'
+import { useParams } from 'react-router-dom'
+import { Diff2HtmlUI } from 'diff2html/lib/ui/js/diff2html-ui-slim.js'
+import 'diff2html/bundles/css/diff2html.min.css'
+import { IconButton } from 'components/common/IconButton/IconButton'
+import { ExpandLeft } from 'components/common/Icons/ExpandLeft'
+import { Hug } from 'components/common/Hug/Hug'
+import { useThemeModeStore } from 'stores/theme-mode'
+import { ColorSchemeType } from 'diff2html/lib/types'
+import { incarnations } from '../../services/incarnations'
+import { useQuery } from '@tanstack/react-query'
+import { useCanShowStatusStore } from 'stores/show-status'
+import { Loader } from 'components/common/Loader/Loader'
+import { Section } from './parts'
+
+export const DiffIncarnation = () => {
+  const diffInjectorRef = useRef<HTMLDivElement>(null)
+  const { id } = useParams()
+  const { mode } = useThemeModeStore()
+  const {
+    isError: isErrorIncarnation,
+    data: incarnationData,
+    isSuccess: isSuccessIncarnation
+  } = useQuery(['incarnations', Number(id)], () => incarnations.getById(id))
+  const { isLoading, isError, data, isSuccess } = useQuery(
+    ['incarnation_diff', Number(id)],
+    () => incarnations.getDiffToTemplate(id)
+  )
+
+  function onBackClick() {
+    window.location.href = '/incarnations/' + id
```

Comment:
Since this is a single page application, I would use `react-router` for page navigation. 
In this case, the `useNavigate` hook would do the trick. 

Also, perhaps a nitpick, but I prefer string interpolation over concatenation
```jsx
window.location.href = `/incarnations/${id}`
```